mlr-org/gallery/optimization/2023-01-23-shadow-variable-search-on-pima/index.qmd

```diff
@@ -0,0 +1,132 @@
+  - classification
+author:
+  - name: Marc Becker
+    url: https://github.com/be-marc
+date: 2023-01-23
+bibliography: ../../bibliography.bib
+knitr:
+  opts_chunk:
+    R.options:
+      datatable.print.nrows: 6
+---
+
+{{< include ../_setup.qmd >}}
+
+```{r 2023-01-23-shadow-variable-search-on-pima-001}
+#| include: false
+
+lgr::get_logger("mlr3")$set_threshold("warn")
+lgr::get_logger("bbotk")$set_threshold("warn")
+set.seed(1)
+future::plan("multisession")
+```
+
+# Scope
+
+Feature selection is the process of finding an optimal set of features to improve the performance and robustness of machine learning algorithms.
+In this article, we introduce the `r ref("mlr3fselect::fselect()")` function and the `r ref("mlr_fselectors_shadow_variable_search", "shadow variable search")` algorithm [@wu_controlling_2007] for a quick and easy feature selection.
+As an example, we will search for the optimal set of features for a `r ref("mlr_learners_classif.svm", "support vector machine")` on the `r ref("mlr_tasks_pima", text = "Pima Indian Diabetes")` data set.
+We assume that you are already familiar with the basic building blocks of the mlr3 ecosystem.
+Some knowledge about `r ref_pkg("mlr3pipelines")` is beneficial but not necessary to understand the example.
+
+# Task and Learner
+
+The objective of the `r ref("mlr_tasks_pima", text = "Pima Indian Diabetes")` data set is to predict whether a person has diabetes or not.
+The data set includes 768 patients with 8 measurements.
+
+```{r 2023-01-23-shadow-variable-search-on-pima-002}
+#| message: false
+
+library(mlr3verse)
+
+task = tsk("pima")
+```
+
+```{r 2023-01-23-shadow-variable-search-on-pima-003}
+#| code-fold: true
+#| column: page
+#| fig-width: 14
+#| fig-height: 4
+#| fig.cap: "Distribution of the features in the Pima Indian Diabetes data set."
+#| warning: false
+
+library(ggplot2)
+task = tsk("pima")
+data = melt(as.data.table(task), id.vars = task$target_names, measure.vars = task$feature_names)
+
+ggplot(data, aes(x = value, fill = diabetes)) +
+  geom_density(alpha = 0.5) +
+  facet_wrap(~ variable, ncol = 8, scales = "free") +
+  scale_fill_viridis_d(end = 0.8) +
+  theme_minimal() +
+  theme(axis.title.x = element_blank())
+```
+
+The data set contains missing values.
+
+```{r 2023-01-23-shadow-variable-search-on-pima-004}
+task$missings()
+```
+
+Support vector machines cannot handle missing values.
+We impute the missing values with the `r ref("mlr_pipeops_imputehist", "histogram imputation")` method.
+
+```{r 2023-01-23-shadow-variable-search-on-pima-005}
+learner = po("imputehist") %>>% lrn("classif.svm", predict_type = "prob")
+```
+
+# Feature Selection
+
+The `r ref("mlr3fselect::fselect()")` function controls and executes the feature selection.
+The function internally creates an `r ref("mlr3fselect::FSelectInstanceSingleCrit", "fselect instance")` and executes the feature selection directly.
+The `method` argument specifies the feature selection algorithm.
```

Comment:
Wie schon mein Kommentar in der introduction: Hier muss zuerst geklärt werden was die grobe struktur von den fselector dingern ist bevor man ueber unterschiedliche Methoden redet. Hier ist noch gar nicht klar was hier abgeht finde ich.